### PR TITLE
Configurable Build

### PR DESCRIPTION
**Pull request by samuelsadok**
Let the user configure the build via the tup.config file.
With this change users should no longer need to edit source files.

### CODE REVIEW

**Comment by madcowswe** [comment]
@samuelsadok Please add an entry to the Changelog.

**Comment by samuelsadok** [comment]
done

File: tup.config.default [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CONFIG_DEBUG=y
+CONFIG_BOARD_VERSION=v3.3
+CONFIG_BOARD_48V=n
```
Comment (madcowswe) [inline tup.config.default]: High voltage version might not always be 48V, I am already thinking of increasing it to 60ish V. So better call it HV.

File: Firmware/Tupfile.lua [hunk 2]
```diff
@@ -1,7 +1,65 @@
...
+    FLAGS += "-DUART_PROTOCOL_LEGACY"
+elseif tup.getconfig("UART_PROTOCOL") == "none" then
+    FLAGS += "-DUART_PROTOCOL_NONE"
+else
+    error("unknown UART protocol "..tup.getconfig("UART_PROTOCOL"))
+end
+
+-- GPIO settings
+if tup.getconfig("STEP_DIR") == "y" then
+    if tup.getconfig("UART_PROTOCOL") != "none" then
```
Comment (madcowswe) [inline Firmware/Tupfile.lua]: Isn't the polarity of this check backwards?

File: Firmware/Tupfile.lua [hunk 3]
```diff
@@ -1,7 +1,65 @@
 
 tup.include('build.lua')
 
-boarddir = 'Board/v3.3'
+-- Switch between board versions
+boardversion = tup.getconfig("BOARD_VERSION")
+if boardversion == "" then boardversion = "v3.4" end
```
Comment (madcowswe) [inline Firmware/Tupfile.lua]: It is dangerous to have a default like this. It hasn't happened yet, but it's concievable that the config of 1 board blows up another board. Better make sure people make it explicit.

File: tup.config.default (hunk 1 above)
Comment (madcowswe) [reply tup.config.default]: Make sure to also update readme.

**Pull request merged**